### PR TITLE
Add documentation to Software Kits

### PR DESCRIPTION
**Pull request by cxhong**

### CODE REVIEW

File: docs/source/advanced/kit/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+Typically, a software kit will contain all of the product package files. However, in some instances, software kits may be delivered as partial or incomplete kits, and not be bundled with all of the product rpms. You will then need to obtain your product packages through your reqular distribution channels, download them to a server along with the partial kit, and run the buildkit addpkgs command to build a complete kit that can be used by xCAT.
+
+
+Contents of a software Kit
+--------------------------
+A software kit is a tar file that contains the following
+
+**Kit Configuration File** --- A file describing the contents of this kit. This file contains following information 
+  * Kit name, version, description, supported OS distributions, license information, and deployment parameters
```
Comment (whowutwut) [inline docs/source/advanced/kit/introduction.rst]: Need a space here, That is why line 19 is all bold.  
image

File: docs/source/advanced/kit/introduction.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+--------------
+Over time it is possible that the details of the Kit package contents and support may change. For example, there may be a need for additional information to be added etc. We refer to a particular instance of the kit support as its "framework". A particular framework is identified by a numerical value.
+
+In order for a kit command to process a kit properly it must be compatible with the level of code that was used to build the kit.
+
+Both the kit commands and the actual kits contain the current framework they support as well as any backlevel versions also supported.
+
+View the supported framework and compatible framework values for a command can be used the "-v|--version" option.
+::
+  addkit -v
```
Comment (whowutwut) [inline docs/source/advanced/kit/introduction.rst]: Suggest adding # before addkit so we can tell which one is the command and which one is the command output

**Comment by bybai** [comment]
@cxhong  Your format for titles are correct in rst format, but we have different third level of title format here: https://xcat-docs.readthedocs.org/en/latest/developers/doc_guidelines.html;
Could you unify the third level of title format as this doc?
We tried in the same third level title in one doc, they appear different size; We unify the format of 4 levels titles, in case some one uses different format in the future.

For example,   
Obtaining the Kits        
^^^^^^^^^^^^^^^^^^^^^^^^^^
To
The Third Title
"`````````````````````"

**Comment by cxhong** [comment]
Thanks,  Yuan.

I modified the changes and committed.

**Comment by bybai** [comment]
I am OK for the doc.

**Comment by cxhong** [comment]
Thanks, Yuan.

**Pull request merged**